code/modules/projectiles/projectile/energy.dm

```diff
@@ -65,21 +66,19 @@
 	icon_state = "dragnetfield"
 	light_range = 3
 	anchored = TRUE
+	var/obj/item/device/radio/beacon/teletarget
 
-/obj/effect/nettingportal/Initialize()
+/obj/effect/nettingportal/Initialize(target)
 	. = ..()
-	var/obj/item/device/radio/beacon/teletarget = null
-	for(var/obj/machinery/computer/teleporter/com in GLOB.machines)
-		if(com.target)
-			if(com.power_station && com.power_station.teleporter_hub && com.power_station.engaged)
-				teletarget = com.target
 
+	teletarget = target
 	addtimer(CALLBACK(src, .proc/pop, teletarget), 30)
 
 /obj/effect/nettingportal/proc/pop(teletarget)
+	to_chat(world, "pop: [teletarget]")//debug
 	if(teletarget)
 		for(var/mob/living/L in get_turf(src))
-			do_teleport(L, teletarget, 2)//teleport what's in the tile to the beacon
+			do_teleport(L, get_turf(teletarget), 2)//teleport what's in the tile to the beacon

```

Comment:
have `get_turf(teletarget)` set to a var and replace this with said var and see if that works.

Reply:
thanks for the tip, but the real problem is that teletarget is `null` according to my little debug messages